packages/kolibri-tools/lib/profile$trs.js

```diff
@@ -0,0 +1,526 @@
+  coach_module: 'CommonCoachStrings',
+  test_component: 'TestComponent',
+};
+
+function profile$trs(localePath, moduleName) {
+  this.localePath = localePath;
+  this.moduleName = moduleName;
+}
+
+/* Webpack Entry */
+profile$trs.prototype.apply = function(compiler) {
+  const self = this;
+
+  // Only works in non-production mode.
+  if (process.env.NODE_ENV !== 'production') {
+    compiler.hooks.emit.tapAsync('Profile$Trs', function(compilation, callback) {
+      let strProfile = getStringDefinitions(self.localePath, self.moduleName);
+
+      // Not all modules have messages files - bail if we don't get one.
+      if (!strProfile) {
+        callback();
+        return;
+      }
+
+      compilation.chunks.forEach(chunk => {
+        let parsedUrl;
+        let ast;
+
+        // Walk through the modules being given to us in compilation.
+        for (const module of chunk.modulesIterable) {
+          parsedUrl = module.resource && url.parse(module.resource);
+
+          // Processing Vue files only here.
+          if (urlIsVue(parsedUrl, module)) {
+            // module._source.source() returns the part of the Vue file between the
+            // <script> tags only - will not parse the <template> tag content at all.
+            ast = espree.parse(module._source.source(), {
+              sourceType: 'module',
+              ecmaVersion: 2018,
+            });
+            strProfile = profileVueScript(strProfile, ast, parsedUrl.pathname, self.moduleName);
+
+            if (parsedUrl.pathname) {
+              // If we have a pathname - which ostensibly points to a Vue file - we
+              // can now parse the <template> portion of the Vue file.
+              const vueFile = fs.readFileSync(parsedUrl.pathname);
+              // Compile the <template>.
+              const template = vueCompiler.compile(vueFile.toString(), {
+                whitespace: 'condense',
+              });
+
+              /**
+               * The vueCompiler.compile() function returns an object including an anemic AST
+               * and a property called `render` which has some stringified JS code.
+               *
+               * That code is what Vue would use to render the component - but it's wrapped in
+               * a `with(this) {}` expression which returns an array of valid JS expressions in
+               * an array.
+               *
+               * It's a string - which is what espree wants anyway - so we can strip the `with()`
+               * altogether, leaving us with a stringified array containing everything we need to
+               * create a thorough AST of the compiled Vue template.               *
+               */
+              const render = template.render.replace(/^.{18}|.{1}$/g, '');
+              ast = espree.parse(render, {
+                ecmaVersion: '2018',
+                sourceType: 'module',
+                ecmaFeatures: { jsx: true, templateStrings: true },
```

Comment:
Worth testing this without the JSX: true - I kind of got squirrely with the options passed to espree while trying to figure out how to get it to parse the render JS without errors.